plugins/cue_points/quiz/lib/kQuizManager.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * @package plugins.quiz
+ * @subpackage quiz
+ */
+class kQuizManager implements kObjectChangedEventConsumer
+{
+	const EVENT_TYPE = 'eventType';
+	const QUIZ_EVENT_TYPE = '@event_type_id@';
+	const KUSER_ID = 'kuserId';
+	const ENTRY_ID = 'entryId';
+	const PARTNER_ID = 'partnerId';
+	const VERSION = 'version';
+	const ANSWER_IDS = 'answerIds';
+	const SCORE = 'score';
+	const NUM_OF_CORRECT_ANSWERS = 'numOfCorrectAnswers';
+	const NUM_OF_QUESTIONS = 'numOfQuestions';
+	const NUM_OF_RELEVANT_QUESTIONS = 'numOfRelevnatQuestions';
+	const CALCULATED_SCORE = 'calculatedScore';
+	const INTERNAL_ANALYTICS_HOST = 'internal_analytics_host';
+	
+	/**
+	 * @param BaseObject $object
+	 * @param array $modifiedColumns
+	 * @return bool
+	 */
+	public function objectChanged (BaseObject $object, array $modifiedColumns)
+	{
+		if (!self::wasStatusChanged($object, $modifiedColumns))
+		{
+			return false;
+		}
+		$quizEventContent = self::getQuizEventContent($object);
+		if (kConf::hasParam(self::INTERNAL_ANALYTICS_HOST))
+		{
+			$statsHost = explode(':', kConf::get(self::INTERNAL_ANALYTICS_HOST));
+			self::sendBeacon(
+				$quizEventContent,
+				$statsHost[0],
+				isset($statsHost[1]) ? $statsHost[1] : 80);
+		}
+		return true;
+	}
+	
+	public static function sendBeacon ($quizEventContent, $host, $port)
+	{
+		requestUtils::sendKavaRequest($host, $port, requestUtils::buildKavaRequest($quizEventContent, $host));
+	}
+	
+	
+	
+	/**
+	 * @param BaseObject $object
+	 * @param array $modifiedColumns
+	 * @return bool
+	 */
+	public function shouldConsumeChangedEvent (BaseObject $object, array $modifiedColumns)
+	{
+		if (self::wasStatusChanged($object, $modifiedColumns))
+		{
+			return true;
+		}
+		return false;
+	}
+	
+	protected static function wasStatusChanged(BaseObject $object, array $modifiedColumns)
+	{
+		return ($object instanceof QuizUserEntry)
+			&& in_array(UserEntryPeer::STATUS, $modifiedColumns)
+			&& $object->getStatus() == QuizPlugin::getCoreValue('UserEntryStatus', QuizUserEntryStatus::QUIZ_SUBMITTED);
+	}
+	
+	protected static function getQuizEventContent($quizUserEntry)
+	{
+		/* @var $quizUserEntry QuizUserEntry */
+		return array(
+			self::EVENT_TYPE => self::QUIZ_EVENT_TYPE,
+			self::KUSER_ID => $quizUserEntry->getkuser()->getId(),
```

Comment:
separate the calls, if we wont get kuser from getKuser you will end up with null exception
add the kuserId only if kuser returned

Reply:
what should be the value if kuser is null?